### ISSUE TITLE
Configuration file not detected or apply in multi root workspace

### ISSUE THREAD

**Issue by fharper** [DavidAnson/vscode-markdownlint#232]
Not sure if it's the issue, but I have a workspace with multiple folders from different path. In one of that folder, I have a `.markdownlint.json` file which work when I use the CLI, but doesn't work with the vscode extension.  I see no errors in the extension output. Any ideas what is happening?

**Comment by DavidAnson** [comment]
Please point to the repository or sketch out the file structure? I can have a look based on that. Thanks!

**Comment by fharper** [comment]
Thanks! This is the repo where the configuration file is: https://github.com/mindee/documentation/

As for my folder structures in my workspace, here it is: it's a multi-root workspace, so each of the root folder are from a totally different path on my computer.
![CleanShot 2022-02-24 at 17 58 07@2x](https://user-images.githubusercontent.com/1170989/155621793-c37303d3-31ab-4d28-9b41-2a04d577f097.png)

**Comment by fharper** [comment]
I highlighter the wrong file on the screenshot btw.

here is the content of `.markdownlint.json`


```
{
	"MD013": false,
    "MD022": false,
	"MD007": {
		"indent": 4
	}
}
```

**Comment by DavidAnson** [comment]
Thanks - I hope to have a look in a week or so.

**Comment by fharper** [comment]
No stress! Let me know if I can help in any way. I can't check the code myself for the next two weeks unfortunately.

**Comment by DavidAnson** [comment]
Can you explain what seems wrong?

To investigate, I've created a multi-root workspace with a clone of the repository you identified within:

```bash
u@C vsc-mdl-232 % cat vsc-mdl-232.code-workspace
{
	"folders": [
		{
			"path": "documentation"
		}
	],
	"settings": {}
}% 
```

I then open the workspace in VS Code via File | Open Workspace From File, update `.markdownlint.json`:

```diff
u@C documentation % git diff
diff --git a/.markdownlint.json b/.markdownlint.json
index 281f158..36a4dce 100644
--- a/.markdownlint.json
+++ b/.markdownlint.json
@@ -1,5 +1,7 @@
 {
-       "MD013": false,
+       "MD013": {
+               "line_length": 40
+       },
     "MD022": false,
        "MD007": {
                "indent": 4
```

And then I open `README.md` and see errors for the customized line length which is what I expect.

**Comment by fharper** [comment]
So I think I found the issue.

If I have a multi-root workspace, with multiple folders, and a configuration file in more than one folder, the extension will take the first one it find, instead of the one in the folder of the file with the focus on.

```shell
Workspace
- Folder One
  - README.md
  - .markdownlint.json
- Folder Two
  - DOCUMENTATION.md
  - .markdownlint.json
```

No matter which markdown file I check, it will always use the `.markdownlint.json` in the ` Folder One`. It make sense for a single root workspace, but in a multi-root workspace, people tend to have multiple projects that may have multiple configurations, so I would expect the extension to go from the directory of the open file, and go backward to find the proper configuration file. Make sense?

**Comment by fharper** [comment]
Actually, my hypothesis may be wrong, sorry. It's the open config file command that always open the first one found in the multi root workspace!

Seems like my problem may not have been a problem after all. I'll close this issue, and will reopen if I have the issue again and can find more information . Sorry about that, and thanks for your time, and the great extension!
